pkg/accountmanager/client.go

```diff
@@ -0,0 +1,58 @@
+package accountmanager
+
+import (
+	"github.com/Optum/dce/pkg/arn"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/iam/iamiface"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+type clienter interface {
+	Config(roleArn *arn.ARN) *aws.Config
+	IAM(roleArn *arn.ARN) iamiface.IAMAPI
+}
+
+// Client helps with client management testing and abstraction
+type client struct {
+	session *session.Session
+	sts     stsiface.STSAPI
+	configs map[string]*aws.Config
+}
+
+// Config configures caching of credentials
+func (c *client) Config(roleArn *arn.ARN) *aws.Config {
+
+	// return no config for nil inputs
+	if roleArn == nil {
+		return nil
+	}
+
+	key := roleArn.String()
+
+	// check for cached config
+	if c.configs != nil && c.configs[key] != nil {
```

Comment:
This caching makes me a little nervous, that we're going to have some invalid cached values somehow.
Given that we're running in Lambda, is this really worth it?

Reply:
The value of these credentials is they auto-renew.  We are storing the thing that manages the auto-renewal.  Helps prevent the system from assuming roles every time we take action.

Reply:
That's what I figured. I just always get anxious about caching these kind of things.